src/h2load.cc

```diff
@@ -271,9 +345,32 @@ void Client::disconnect() {
 }
 
 void Client::submit_request() {
+  restart_timeout();
   auto req_stat = &worker->stats.req_stats[worker->stats.req_started++];
   session->submit_request(req_stat);
   ++req_started;
+
+  // if an active timeout is set and this is the last request to be submitted
+  // on this connection, start the active timeout.
+  if (worker->config->conn_active_timeout > 0 && req_started >= req_todo) {
+    ev_timer_start(worker->loop, &conn_active_watcher);
+  }
+}
+
+void Client::process_timedout_streams() {
+  for (auto req_stat : worker->stats.req_stats) {
+    if (!req_stat.completed) {
+      req_stat.stream_close_time = std::chrono::steady_clock::now();
+    }
+  }
+  auto req_timed_out = req_todo - req_done;
+
+  worker->stats.req_failed += req_timed_out;
+  worker->stats.req_error += req_timed_out;
+  worker->stats.req_done += req_timed_out;
+  worker->stats.req_timedout += req_timed_out;
+
+  req_done = req_todo;
```

Comment:
Please use :process_abandoned_streams() to update worker->stats except for req_timeout.